src/Instrumentation/ExtRdKafka/tests/Integration/ExtRdKafkaInstrumentationTest.php

```diff
@@ -0,0 +1,218 @@
+        $this->assertCount(2, $this->storage);
+
+        // When
+        // This should create a third span, for consumption, which should have the same TraceID as the first span
+        $this->consumeMessage();
+
+        // Then
+        $this->assertCount(3, $this->storage);
+        /** @var ImmutableSpan $span */
+        $span = $this->storage->offsetGet(2);
+        $this->assertEquals('test process', $span->getName());
+        $this->assertEquals($expectedTraceId, $span->getContext()->getTraceId());
+    }
+
+    public function test_context_set_in_kafka_headers_on_message_production(): void
+    {
+        // Given
+        $this->assertCount(0, $this->storage);
+        $tracerProvider = Globals::tracerProvider();
+        $tracer = $tracerProvider->getTracer('test');
+        $span = $tracer->spanBuilder('test_span')->startSpan();
+        $scope = $span->activate();
+
+        $this->produceMessage('test');
+
+        $scope->detach();
+        $span->end();
+
+        // When
+        $message = $this->consumeMessage();
+
+        $this->assertInstanceOf(Message::class, $message);
+        $this->assertIsArray($message->headers);
+        $this->assertArrayHasKey('traceparent', $message->headers);
+
+        // The traceparent header is separated into 4 sections with dashes. I want to get the second segment of that
+        // header, which is the TraceID
+        $traceId = explode('-', $message->headers['traceparent'])[1];
+        $this->assertEquals($span->getContext()->getTraceId(), $traceId);
+    }
+
+    public function test_produce_creates_new_span()
+    {
+        $this->assertCount(0, $this->storage);
+        $this->produceMessage('test');
+
+        $this->assertCount(1, $this->storage);
+
+        $span = $this->storage->offsetGet(0);
+        $this->assertEquals('test publish', $span->getName());
+    }
+
+    private function produceMessage(
+        string $message,
+        ?string $key = null,
+        array $headers = null,
+        bool $produceWithoutHeaders = false
+    ): void {
+        $conf = new Conf();
+        $producer = new Producer($conf);
+        $producer->addBrokers(getenv('KAFKA_HOST') ?: 'localhost' . ':9092');
+
+        $topic = $producer->newTopic(self::STANDARD_TOPIC_NAME);
+
+        if ($produceWithoutHeaders) {
+            $topic->produce(RD_KAFKA_PARTITION_UA, 0, $message, $key);
+        } else {
+            $topic->producev(RD_KAFKA_PARTITION_UA, 0, $message, $key, $headers);
+        }
+        $producer->poll(100);
+        for ($flushRetries = 0; $flushRetries < 10; $flushRetries++) {
+            $result = $producer->flush(10000);
+            if (RD_KAFKA_RESP_ERR_NO_ERROR === $result) {
+                break;
+            }
+        }
+    }
+
+    /**
+     * @psalm-suppress InvalidNullableReturnType
+     * @return int|Message
+     *
+     * Couldn't use union types on method as psalm wants php 7.4 compatible code. For some reason, it also thinks
```

Comment:
that doesn't sound right - is it somehow looking at the top-level composer.json to determine php version? if all else fails, you can force it in the local psalm.xml.dist: https://psalm.dev/docs/running_psalm/configuration/#phpversion

Reply:
Found it in the logs- I wasn't specifying PHP Version when doing the `make all`- as tests didn't require it. Fixed now